pkg/certificate/interface.go

```diff
@@ -0,0 +1,101 @@
+package certificate
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	Auto        ProviderType = "auto"
+	CertManager ProviderType = "cert-manager"
+	// add more ...
+)
+
+func NewProvider(pt ProviderType) Provider {
+	switch pt {
+	case Auto:
+		return nil // change me later
+	case CertManager:
+		return nil // change me later
+	}
+	panic(fmt.Sprintf("unknown provider type: %s", pt))
+}
+
+type Config struct {
+	CommonName       string
+	Organization     []string
+	DNSNames         []string
+	ValidityDuration time.Duration
+	CA               []byte
```

Comment:
`CABundle` maybe, as commented on the doc.
and it can still be a bundle of 1 CA.

then again, most of the flags of k8s components use just `ca` that can actually be a bundle.

Reply:
Renamed to CABundle & thx